internal/pkg/agent/install/uninstall.go

```diff
@@ -310,3 +318,41 @@ func applyDynamics(ctx context.Context, log *logger.Logger, cfg *config.Config)
 
 	return config.NewConfigFrom(finalConfig)
 }
+
+// killWatcher finds and kills any running Elastic Agent watcher.
+func killWatcher() error {
+	procStats := process.Stats{
+		// filtering with '.*elastic-agent' or '^.*elastic-agent$' doesn't
+		// seem to work as expected, filtering is done in the for loop below
+		Procs: []string{".*"},
+	}
+	err := procStats.Init()
+	if err != nil {
+		return fmt.Errorf("failed to initialize process.Stats: %w", err)
+	}
+	pidMap, _, err := procStats.FetchPids()
+	if err != nil {
+		return fmt.Errorf("failed to fetch pids: %w", err)
+	}
+	var errs error
+	for pid, state := range pidMap {
+		if len(state.Args) < 2 {
+			// must have at least 2 args "elastic-agent[.exe] watch"
+			continue
+		}
+		if filepath.Base(state.Args[0]) == paths.BinaryName && state.Args[1] == "watch" {
+			// it is the watch subprocess
+			proc, err := os.FindProcess(pid)
+			if err != nil {
+				errs = errors.Join(errs, fmt.Errorf("failed to load watcher process with pid %d: %w", pid, err))
+				continue
+			}
+			err = proc.Kill()
```

Comment:
> // Kill causes the Process to exit immediately. Kill does not wait until
> // the Process has actually exited. This only kills the Process itself,
> // not any other processes it may have started.
> func (p *Process) Kill() error {

The docs on this say it doesn't actually wait for the process to exit, which suggests there is a small race condition here with any following attempt to install depending on how long this actually takes. On Unix it is probably almost instant, not sure how long it takes on Windows.

Reply:
Yeah I am testing this on Windows with my integration runner.

I am still seeing `Error: failed to remove installation directory (C:\\Program Files\\Elastic\\Agent): could not remove path, timeout exeeded after 69 tries during 7s. Last error: remove C:\\Program Files\\Elastic\\Agent\\data\\elastic-agent-162ea9\\logs\\elastic-agent-watcher-20230908-1.ndjson: The process cannot access the file because it is being used by another process.\nFor help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.11/fleet-troubleshooting.html\n`.

Could be that the `Kill` is not instant on Windows, but I also read that waiting for a process that is not your child doesn't always work. Maybe I should add a loop in this function to keep calling `Kill` on all found processes until they are all gone.

Reply:
Yeah I think you will have to poll waiting for the process to exit.

Reply:
I added the loop to check for the watcher, this ensures that the watcher is fully killed before continuing.